src/AssetRegistry/MainRegistry.sol

```diff
@@ -103,36 +102,73 @@ contract MainRegistry is Ownable {
      * - WETH is a native token for ETH as BaseCurrency
-     * @dev The list of Credit Rating Categories should or be as long as the number of assets added to the Main Registry,
-     * or the list must have length 0. If the list has length zero, the credit ratings of the baseCurrency for all assets
-     * is initiated as credit rating with index 0 by default (worst credit rating).
-     * Each Credit Rating Category is labeled with an integer, Category 0 (the default) is for the most risky assets.
-     * Category from 1 to 9 will be used to label groups of assets with similar risk profiles
-     * (Comparable to ratings like AAA, A-, B... for debtors in traditional finance).
+     * @dev The list of Risk Variables (Collateral Factor and Liquidation Threshold) should or be as long as
+     * the number of assets added to the Main Registry,or the list must have length 0.
+     * If the list has length zero, the risk variables of the baseCurrency for all assets
+     * is initiated as default (safest lowest rating).
+     * @dev Risk variable have 2 decimals precision
      */
     function addBaseCurrency(
         BaseCurrencyInformation calldata baseCurrencyInformation,
-        uint256[] calldata assetCreditRatings
+        uint16[] calldata baseCurrencyCollateralFactors,
+        uint16[] calldata baseCurrencyLiquidationThresholds
     ) external onlyOwner {
         baseCurrencyToInformation[baseCurrencyCounter] = baseCurrencyInformation;
         assetToBaseCurrency[baseCurrencyInformation.assetAddress] = baseCurrencyCounter;
         isBaseCurrency[baseCurrencyInformation.assetAddress] = true;
         baseCurrencies.push(baseCurrencyInformation.assetAddress);
 
-        uint256 assetCreditRatingsLength = assetCreditRatings.length;
+        // Check: Valid length of arrays
+        uint256 baseCurrencyCollateralFactorsLength = baseCurrencyCollateralFactors.length;
         require(
-            assetCreditRatingsLength == assetsInMainRegistry.length || assetCreditRatingsLength == 0, "MR_AN: length"
+            (
+                baseCurrencyCollateralFactorsLength == assetsInMainRegistry.length
+                    && baseCurrencyCollateralFactorsLength == baseCurrencyLiquidationThresholds.length
+            ) || (baseCurrencyCollateralFactorsLength == 0 && baseCurrencyLiquidationThresholds.length == 0),
+            "MR_ABC: LENGTH_MISMATCH"
         );
-        for (uint256 i; i < assetCreditRatingsLength;) {
-            require(assetCreditRatings[i] < CREDIT_RATING_CATOGERIES, "MR_AN: non existing credRat");
-            assetToBaseCurrencyToCreditRating[assetsInMainRegistry[i]][baseCurrencyCounter] = assetCreditRatings[i];
+        // Logic Fork: If the list are empty, initate the variables with default collateralFactor and liquidationThreshold
+        if (baseCurrencyCollateralFactorsLength == 0) {
+            // Loop: Per base currency
+            for (uint256 i; i < assetsInMainRegistry.length;) {
```

Comment:
could be in separate var, but if we inverse the coll thresholds etc this will be removed anyway